Update release-trueos.sh

beanpole135 opened the pull request:
Update the essential packages checks a bit:
1. If any essential packages are missing, print out the full list of package names which are missing at the end of the routine.
2. Add a new JSON manifest option: "iso" -> "optional-dist-packages". This is treated the same as the "dist-packages" lists, but are not used when doing the essential package checks (optional packages will not fail the build if they are not available).

beanpole135 committed:
Update release-trueos.sh

Update the essential packages checks a bit:
1. If any essential packages are missing, print out the full list of package names which are missing at the end of the routine.
2. Add a new JSON manifest option: "iso" -> "dist-packages-are-essential" : [true/false]. This determines whether the list of dist-packages should be included in the essential package checks (default: true).

beanpole135 committed:
Update README.md

Also add the new "iso"."dist-packages-are-essential" option to the README for the manifest formats.

kmoore134 commented:
Dist packages are already considered essential. Why would you want them not to be?

beanpole135 commented:
The iso-packages and auto-install-packages should definitely be considered essential all the time, but the dist packages list for Trident includes a lot of optional-install packages which we could include or not on the ISO - they are not crucial to us just nice to have installed out of box for some users.
In that case, we already have our "essential" packages listed in ports -> build and iso -> iso-packages, but the dist packages list includes a lot of optional ones that should not fail the build if they are not there, but should still be added to the ISO if they are.

beanpole135 commented:
As an example: we have the "hplip" printer driver package included in the dist-files for the ISO so that people can have that installed out-of-box as needed. That package fails to build with the current ports tree (net-snmp dependency failing), but we do not want that package to be considered essential and stop the build for us.

kmoore134 commented:
I'm not sure I agree with that logic. That means you can have wildly inconsistent iso files getting produced. In my mind if you want that, you should add some optional-dist directive, but again even that sounds like a huge POLA violation since you'll have random results

beanpole135 commented:
Effectively, that option allows the dist-packages directive to *become* the optional-dist directive. The essentials are already specified in the other "iso" package fields and the "dist-packages" field becomes the list of optional ones when that flag is enabled.
Again - current behavior will not be changes by the addition of this flag. It just provides the option to treat that field an alternate way *if* specifically flagged that way.

beanpole135 commented:
If you like, I will re-work this change to enable the use of a new "optional-dist-packages" field instead of changing the behavior of the existing field though.

kmoore134 commented:
I'd prefer it be broken out into is own section, so we know 100% for sure that things listed in dist-packages are required. I don't want weird cases were somebody set the flag wrong and all the sudden got ISO's with missing packages. Want that kind of determinism.

beanpole135 committed:
Update README.md

Change the readme to use an "optional-dist-packages" field/list instead of the "dist-packages-are-essential" flag.

beanpole135 committed:
Update release-trueos.sh

Change the "dist-packages-are-essential" flag into supporting a new "optional-dist-packages" list instead. This makes it easier to identify which dist packages are essential, and which ones are optional.
Also add build output notices about which optional packages are missing (when found) and a summary list at the end of all the package copies.

beanpole135 commented:
ok - everything switched over to supporting a new "optional-dist-packages" field instead of the "dist-packages-are-essential" field I originally wrote up.
